File: Loenn/entities/recharge_platform.lua
local utils = require("utils")

local recharge_platform = {}

recharge_platform.name = "batteries/recharge_platform"

recharge_platform.placements = {name = "default"}

recharge_platform.texture = "batteries/recharge_platform/base0"
recharge_platform.justification = {0.5, 1}

function recharge_platform.selection(room, entity)
    return utils.rectangle(entity.x - 11, entity.y - 4, 22, 4)
end

return recharge_platform

File: Ahorn/entities/recharge_platform.jl
module BatteriesRechargePlatform

using ..Ahorn, Maple

@mapdef Entity "batteries/recharge_platform" BatteryRechargePlatform(x::Integer, y::Integer)

const placements = Ahorn.PlacementDict(
    "Battery Recharge Platform (Batteries)" => Ahorn.EntityPlacement(
        BatteryRechargePlatform
    )
)

sprite = "batteries/recharge_platform/base0"

function Ahorn.selection(entity::BatteryRechargePlatform)
    x, y = Ahorn.position(entity)

    return Ahorn.getSpriteRectangle(sprite, x, y - 4)
end

Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::BatteryRechargePlatform, room::Maple.Room) = Ahorn.drawSprite(ctx, sprite, 0, -4)

end
